client/src/client/workers/test-evaluator.ts

```diff
@@ -68,14 +85,15 @@ self.onmessage = async e => {
   const assert = chai.assert;
   const __helpers = curriculumHelpers;
   // Fake Deep Equal dependency
-  const DeepEqual = (a, b) => JSON.stringify(a) === JSON.stringify(b);
+  const DeepEqual = <T>(a: T, b: T) => JSON.stringify(a) === JSON.stringify(b);
 
   // Build errors should be reported, but only once:
   __utils.toggleProxyLogger(e.data.firstTest);
-  /* eslint-enable no-unused-vars */
+  /* eslint-enable @typescript-eslint/no-unused-vars */
   try {
     let testResult;
-    let __userCodeWasExecuted = false;
+    const __userCodeWasExecuted = false;
```

Comment:
```suggestion
    let __userCodeWasExecuted = false;
```
This can be reassigned by the eval inside the try block, so it should be declared as a `let`